### ISSUE TITLE
Http4k-testing-approval should support binary comparisons

### ISSUE THREAD

**Issue by oharaandrew314** [http4k/http4k#1183]
Currently, if you use `ApprovalTest` for binary content, the `.actual` file will be coerced to text and corrupted.  However, if you place the actual expected binary and give it the correct `.approved` file name, the test will pass.

This isn't ideal.  It would be nice if the generated `.approved` file was valid.

**Comment by oharaandrew314** [comment]
It seems most of the work is already done.  Just need to add this.

```kotlin
class BinaryApprovalTest : BaseApprovalTest {
    override fun approverFor(context: ExtensionContext): Approver = NamedResourceApprover(
        ClassAndMethod.nameFor(context.requiredTestClass, context.requiredTestMethod),
        HttpBinaryBody(),
        FileSystemApprovalSource(File("src/test/resources"))
    )
}
```

**Comment by daviddenton** [comment]
Fixed in v5.32.3.0
